<a href="https://colab.research.google.com/github/andiunsia/DataScience_240401010009_Andi/blob/main/Pertemuan2_Andi_240401010009.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

---

Nama : Andi

NIM : 240401010009

Kelas : IF401

---

# Latihan Pertemuan 2

In [ ]:
import pandas as pd
import numpy as np
# ① Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

# ② Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

# ③ Filtering — Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
(df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

# ④ Groupby — Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
total=('PassengerId','count'),
selamat=('Survived','mean'),
).round(3)
print(hasil)

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
0.9680851063829787 34.61176470588235
        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


# Pertanyaan 1
## Berapa jumlah baris dan kolom dataset Titanic?

In [ ]:
# Pertanyaan 1
# Berapa jumlah baris dan kolom dataset Titanic?

df_shape = df.shape
baris = df_shape[0]
kolom = df_shape[1]
print('Jumlah baris:', baris)
print('Jumlah kolom:', kolom)

Jumlah baris: 891
Jumlah kolom: 12


# Pertanyaan 2
## Kolom mana yang memiliki nilai kosong (missing)?


In [ ]:
# Pertanyaan 2
# Kolom mana yang memiliki nilai kosong (missing)?

missing_colm = df.isnull().sum()
missing_df = pd.DataFrame({
    'Jumlah Missing': missing_colm,
})

# Kolom yang memiliki missing
print(missing_df[missing_df['Jumlah Missing'] > 0])

          Jumlah Missing
Age                  177
Cabin                687
Embarked               2


# Pertanyaan 3
## Berapa persen penumpang yang selamat (Survived=1)?

In [ ]:
# Pertanyaan 3
# Berapa persen penumpang yang selamat (Survived=1)?

df_survived = df[df['Survived'] == 1]
survived_count = df_survived.shape[0]
total_count = df.shape[0]
print('Penumpang yang selamat sebanyak',(survived_count / total_count) * 100,'%')

Penumpang yang selamat sebanyak 38.38383838383838 %


# Pertanyaan 4
## Rata-rata usia penumpang wanita kelas 1 yang selamat?

In [ ]:
# Rata-rata usia penumpang wanita kelas 1 yang selamat?

df_female_class1_survived = df[(df['Sex'] == 'female') & (df['Pclass'] == 1) & (df['Survived'] == 1)]
df_female_class1_survived_age_series = df_female_class1_survived['Age']
print('Rata-rata berusia',df_female_class1_survived_age_series.mean(),'Tahun')

Rata-rata berusia 34.9390243902439 Tahun


#Pertanyaan 5
## Kelas penumpang mana yang tingkat keselamatan tertinggi?

In [ ]:
# Kelas penumpang mana yang tingkat keselamatan tertinggi?

class_survive = df.groupby('Pclass')['Survived'].mean()
persen = class_survive.max()*100
print('Kelas dengan kelesamatan tertinggi adalah kelas ',class_survive.idxmax(),'dengan',persen,'% penumpang selamat')


Kelas dengan kelesamatan tertinggi adalah kelas  1 dengan 62.96296296296296 % penumpang selamat
Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


# Pertanyaan 6
## Tampilkan hasil groupby

In [ ]:
# Tampilkan hasil groupby
print(df.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64




---


# Kesimpulan

**Eksplorasi Data Titanic dengan Pandas**

Tujuan eksplorasi data adalah memahami struktur data, menangani nilai hilang, dan menggali pola keselamatan penumpang. Berikut rangkuman hasil analisis:

**- Struktur Dataset**

Dataset memiliki 891 baris dan 12 kolom. Kolom yang tersedia meliputi identitas penumpang, kelas, jenis kelamin, usia, jumlah saudara/orang tua, tiket, tarif, kabin, dan status selamat.

**- Identifikasi Nilai Hilang**

Terdapat tiga kolom dengan nilai kosong dan ini perlu ditangani sebelum analisis lebih lanjut, misalnya dengan imputasi atau penghapusan.

**- Tingkat Keselamatan Umum**

Secara keseluruhan, hanya 38,38% penumpang yang selamat.

**- Profil Penumpang Wanita Kelas 1 yang Selamat**

Rata-rata usia mereka adalah 34,94 tahun, dengan tingkat keselamatan mencapai 96,8%.

**- Tingkat Keselamatan Berdasarkan Kelas**

Terdapat perbedaan signifikan antar kelas:

*   Kelas 1: 62,96%

*   Kelas 2: 47,28%

*   Kelas 3: 24,24%


**- Keterampilan Praktis dengan Pandas**

Latihan ini menggunakan fungsi-fungsi dasar Pandas, seperti:

*   read_csv() untuk memuat data,

*   shape, isnull().sum() untuk inspeksi data,

*   filtering dengan kondisi majemuk,

*   groupby().agg() untuk agregasi statistik,

*   mean(), idxmax() untuk analisis numerik.



